python/dolfinx/geometry.py

```diff
@@ -270,3 +270,46 @@ def compute_distance_gjk(
 
     """
     return _cpp.geometry.compute_distance_gjk(p, q)
+
+
+def determine_point_ownership(
+    mesh: Mesh,
+    points: npt.NDArray[np.floating],
+    cells: typing.Optional[npt.NDArray[np.int32]] = None,
+    padding: float = 0.0,
+) -> PointOwnershipData:
+    """Build point ownership data for a mesh-points pair.
+
+    First, potential collisions are found by computing intersections
+    between the bounding boxes of the cells and the set of points.
+    Then, actual containment pairs are determined using the GJK algorithm.
+
+    Args:
+        mesh: The mesh
+        points: Points to check for collision (``shape=(num_points, gdim)``)
+        cells: Cells to check for ownership
+            If ``None`` then all cells are considered.
+        padding: Amount of absolute padding of bounding boxes of the mesh.
+        Each bounding box of the mesh is padded with this amount, to increase
+        the number of candidates, avoiding rounding errors in determining the owner
+        of a point if the point is on the surface of a cell in the mesh.
+
+    Returns:
+        Point ownership data
```

Comment:
Are the fields of PointOwnershipData documented in the Python wrapped class?

Reply:
https://github.com/FEniCS/dolfinx/blob/6fda65d339a32cea2972ae89e9dfbde5339d6320/python/dolfinx/geometry.py#L47-L61

The doc is the same as on the c++ side.

Reply:
Looks good, thanks for pointing it out to me.

The C++ documentation for the return argument describing PointOwnershipData looks over-specified given the struct is now properly documented.

Reply:
Removed most of it.